src/app/complex/complex-results/complex-results.component.ts

```diff
@@ -41,44 +48,45 @@ export class ComplexResultsComponent implements OnInit, AfterViewInit {
 
   ngOnInit() {
     this.titleService.setTitle('Complex Portal - Results');
-    this._allInteractorsInComplexSearch = [];
-    this.route.fragment.subscribe(fragment => {
+    this.allInteractorsInComplexSearch = [];
+    this.route.fragment.pipe(take(1)).subscribe(fragment => {
       if (fragment === COMPLEX_NAVIGATOR_VIEW) {
-        this.DisplayType = COMPLEX_NAVIGATOR_VIEW;
-      } else if (fragment === LIST_VIEW) {
-        this.DisplayType = LIST_VIEW;
+        this.displayType = COMPLEX_NAVIGATOR_VIEW;
+      } else {
+        this.displayType = LIST_VIEW;
       }
-      this.route
```

Comment:
There was a reason `this.route.queryParams.subscribe` was inside `this.route.fragment.subscribe`. 

`this.requestComplexResults();` calls the back-end with a param for the page size. The page size depends on the view mode (list or navigator), which may depend on the fragment set.

By having the subscribe one inside the other, we make sure if there is a fragment, it is set properly before making the back-end call. If fragment is null, `this.route.fragment.subscribe` still works, I tested that.

Now there are 2 separate subscribe calls that can happen in any order.

Also, we had
```
      if (fragment === COMPLEX_NAVIGATOR_VIEW) {
        this.DisplayType = COMPLEX_NAVIGATOR_VIEW;
      } else if (fragment === LIST_VIEW) {
        this.DisplayType = LIST_VIEW;
      }
```
instead of 
```
      if (fragment === COMPLEX_NAVIGATOR_VIEW) {
        this.displayType = COMPLEX_NAVIGATOR_VIEW;
      } else {
        this.displayType = LIST_VIEW;
      }
```
because list view is not the default view. If the fragment is specified, then we use that, if not, when we process the search results, based on the number of results, we set one or the other.

Reply:
Basically the problem was that if the fragment is null, then displayType is null which made that no button was selected, even though list view was being rendered. In general I think it's better practice to avoid null values in this case, but I'm fine for this case if you think it's really better. 

Regarding the nested subsribes, I don't think that's a good thing as you will create many subsrctiptions which will never be unsubsrcribed. I'm switching it to a fragment -> tap -> switchMap(queryParam) -> subsribe. This way only one subscribe, and basically you retrigger the requestCompelxResults() every time the fragment change, after having set the right view with the tap.

However, regarding the setting of the view according to the number of results, I couldn't find this logic, and I don't think it exists, otherwise, we wouldn't have null value for displayType.

Reply:
See [2cf650f](https://github.com/Complex-Portal/complex-portal-view/pull/225/commits/2cf650f5f97f110705c9369e2d845f4fa21edae6) to consider if the changes suits you or not

Reply:
It happens in `processSearchResults`.

If we have no filters and there's 1 result, then we redirect to the single complex page.
Else, if displayType is not set (that's why it can be null initially), if we have less or equal to 20 results, then displayType is set as navigator, otherwise as list.

If the fragment is specified, then displayType is set from the fragment, if it's not, it will be based on the number of results.

Reply:
Somehow it's not working and it's never setting the default view now. It's always list view by default.

Reply:
I've fixed it locally. I actually broke it some months ago...

In `complex-list-display-buttons.component.ts`:
```
displayType = model<SearchDisplay>();
```

In `complex-results.component.ts`:
- Delete line 218
   ```
   this.displayType = displayType;
   ```
- Add to the beginning of `setListView()`
   ```
   this.displayType = SearchDisplay.list;
   ```
- Add to the beginning of `setComplexNavigatorView()`
   ```
   this.displayType = SearchDisplay.navigator;
   ```

Reply:
Done in [c6baad2](https://github.com/Complex-Portal/complex-portal-view/pull/225/commits/c6baad2446f15a17111cada2dcb6c9d33e9992d3)